File: app/src/main/java/com/sachtech/datingapp/ui/profile/LanguageDialog.java
// LanguageDialog.java
package com.sachtech.datingapp.ui.profile;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import com.sachtech.datingapp.R;
import com.sachtech.datingapp.R.id;
import com.sachtech.datingapp.ui.profile.listener.OnLanguageSelect;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.HashMap;

import kotlin.jvm.internal.Intrinsics;

public final class LanguageDialog extends DialogFragment {
  OnLanguageSelect onLanguageSelect;
   @NotNull
   private String language = "";
   @NotNull
   public ArrayList list;
   private HashMap _$_findViewCache;

   @Nullable
   public final OnLanguageSelect getOnLanguageSelect() {
      return this.onLanguageSelect;
   }

   public final void setOnLanguageSelect(@Nullable OnLanguageSelect var1) {
      this.onLanguageSelect = var1;
   }

   public boolean adjustdisplay() {
      return true;
   }

   @NotNull
   public final String getLanguage() {
      return this.language;
   }

   public final void setLanguage(@NotNull String var1) {
      Intrinsics.checkParameterIsNotNull(var1, "<set-?>");
      this.language = var1;
   }

   @androidx.annotation.Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @androidx.annotation.Nullable ViewGroup container, @androidx.annotation.Nullable Bundle savedInstanceState) {
      return LayoutInflater.from(getContext()).inflate(R.layout.dialog_language,container,false);
   }

   public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
      super.onViewCreated(view, savedInstanceState);
      getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
      list = new ArrayList();
      TextView languageAdd = view.findViewById(id.langugageAdd);
      EditText language1 = view.findViewById(id.language1);
      EditText language2 = view.findViewById(id.language2);
      languageAdd.setOnClickListener(new OnClickListener() {
         @Override
         public void onClick(View v) {
            if (!language1.getText().toString().isEmpty() && !language2.getText().toString().isEmpty()) {
               language = "" + language1.getText().toString()+" "+language2.getText().toString();
               list.add(language1.getText().toString());
               list.add(language2.getText().toString());
               onLanguageSelect.getLanguage(list);
               dismiss();
            } else if (!language1.getText().toString().isEmpty()) {
               language = "" + language1.getText().toString();
               list.add(language);
               onLanguageSelect.getLanguage(list);
               dismiss();
            } else if (!language2.getText().toString().isEmpty()) {
               language = "" + language2.getText().toString();
               list.add(language);
               onLanguageSelect.getLanguage(list);
               dismiss();
            }
         }
      });
   }

   public final void languageSelectListener(@NotNull OnLanguageSelect onLanguageSelect) {
      this.onLanguageSelect = onLanguageSelect;
   }



}

File: app/src/main/java/com/sachtech/datingapp/ui/profile/LanguageDialog.kt
/*
package com.sachtech.datingapp.ui.profile

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View
import androidx.fragment.app.DialogFragment
import com.sachtech.datingapp.R
import gurtek.mrgurtekbase.KotlinBaseDialogFragment
import kotlinx.android.synthetic.main.dialog_language.*

class LanguageDialog : KotlinBaseDialogFragment<DialogFragment>(R.layout.dialog_language) {

    var onLanguageSelect: OnLanguageSelect? = null
    override fun adjustdisplay(): Boolean {
        return true
    }

    var language = ""
    lateinit var list: ArrayList<String>
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        dialog?.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        list = ArrayList()
        langugageAdd.setOnClickListener {
            if (language1.text.isNotEmpty() && language2.text.isNotEmpty()) {
                language = "${language1.text},${language2.text}"
                list.add(language1.text.toString())
                list.add(language2.text.toString())
                onLanguageSelect?.getLanguage(list)
                dismiss()
            } else if (language1.text.isNotEmpty()) {
                language = "${language1.text}"
                list.add(language)
                onLanguageSelect?.getLanguage(list)
                dismiss()
            } else if (language2.text.isNotEmpty()) {
                language = "${language2.text}"
                list.add(language)
                onLanguageSelect?.getLanguage(list)
                dismiss()
            }
        }
    }

    fun languageSelectListener(onLanguageSelect: OnLanguageSelect) {
        this.onLanguageSelect = onLanguageSelect
    }
}

interface OnLanguageSelect {
    fun getLanguage(language: ArrayList<String>?)
}*/
